webapp/src/ts/modules/tasks/tasks.component.ts

```diff
@@ -116,52 +128,86 @@ export class TasksComponent implements OnInit, OnDestroy {
     window.location.reload();
   }
 
-  private hydrateEmissions(taskDocs) {
+  private async hydrateEmissions(taskDocs) {
     return taskDocs.map(taskDoc => {
       const emission = { ...taskDoc.emission };
       const dueDate = moment(emission.dueDate, 'YYYY-MM-DD');
       emission.date = new Date(dueDate.valueOf());
       emission.overdue = dueDate.isBefore(moment());
       emission.owner = taskDoc.owner;
-
       return emission;
     });
   }
 
-  private refreshTasks() {
-    const telemetryData:any = {
-      start: Date.now(),
-    };
-
-    return this.rulesEngineService
-      .isEnabled()
-      .then(isEnabled => {
-        this.tasksDisabled = !isEnabled;
-        return isEnabled ? this.rulesEngineService.fetchTaskDocsForAllContacts() : [];
-      })
-      .then(taskDocs => {
-        this.hasTasks = taskDocs.length > 0;
-        this.loading = false;
-        this.tasksActions.setTasksList(this.hydrateEmissions(taskDocs));
-        if (!this.tasksLoaded) {
-          this.tasksActions.setTasksLoaded(true);
-        }
-
-        telemetryData.end = Date.now();
-        const telemetryEntryName = !this.tasksLoaded ? `tasks:load`: `tasks:refresh`;
-        this.telemetryService.record(telemetryEntryName, telemetryData.end - telemetryData.start);
-      })
-      .catch(err => {
-        console.error('Error getting tasks for all contacts', err);
-
-        this.error = true;
-        this.loading = false;
-        this.hasTasks = false;
-        this.tasksActions.setTasksList([]);
-      });
+  async refreshTasks() {
```

Comment:
Since this is not called in the DOM, then it can remain private as before.
```suggestion
  private async refreshTasks() {
```